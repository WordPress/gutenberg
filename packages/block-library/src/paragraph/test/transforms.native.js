/**
 * External dependencies
 */
import {
	getEditorHtml,
	initializeEditor,
	setupCoreBlocks,
	transformBlock,
	getBlockTransformOptions,
} from 'test/helpers';

const block = 'Paragraph';
const initialHtml = `
<!-- wp:paragraph -->
<p>Example text</p>
<!-- /wp:paragraph -->`;

// NOTE: Paragraph block can be transformed to Buttons block in web,
// however this transform is not supported in the native version.
const transformsWithInnerBlocks = [ 'List', 'Quote', 'Columns', 'Group' ];
const blockTransforms = [
	'Heading',
	'Preformatted',
	'Pullquote',
	'Verse',
	'Code',
	...transformsWithInnerBlocks,
];

setupCoreBlocks();

describe( `${ block } block transforms`, () => {
	test.each( blockTransforms )( 'to %s block', async ( blockTransform ) => {
		const screen = await initializeEditor( { initialHtml } );
		const newBlock = await transformBlock( screen, block, blockTransform, {
			hasInnerBlocks:
				transformsWithInnerBlocks.includes( blockTransform ),
		} );
		expect( newBlock ).toBeVisible();
		expect( getEditorHtml() ).toMatchSnapshot();
	} );

	it( 'matches expected transformation options', async () => {
		const screen = await initializeEditor( { initialHtml } );
		const transformOptions = await getBlockTransformOptions(
			screen,
			block
		);
		expect( transformOptions ).toHaveLength( blockTransforms.length );
	} );
} );
