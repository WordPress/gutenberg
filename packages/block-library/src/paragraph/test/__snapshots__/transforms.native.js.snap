// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Paragraph block transforms to Code block 1`] = `
"<!-- wp:code -->
<pre class="wp-block-code"><code>Example text</code></pre>
<!-- /wp:code -->"
`;

exports[`Paragraph block transforms to Columns block 1`] = `
"<!-- wp:columns -->
<div class="wp-block-columns"><!-- wp:column {"width":"100%"} -->
<div class="wp-block-column" style="flex-basis:100%"><!-- wp:paragraph -->
<p>Example text</p>
<!-- /wp:paragraph --></div>
<!-- /wp:column --></div>
<!-- /wp:columns -->"
`;

exports[`Paragraph block transforms to Group block 1`] = `
"<!-- wp:group {"layout":{"type":"constrained"}} -->
<div class="wp-block-group"><!-- wp:paragraph -->
<p>Example text</p>
<!-- /wp:paragraph --></div>
<!-- /wp:group -->"
`;

exports[`Paragraph block transforms to Heading block 1`] = `
"<!-- wp:heading -->
<h2 class="wp-block-heading">Example text</h2>
<!-- /wp:heading -->"
`;

exports[`Paragraph block transforms to List block 1`] = `
"<!-- wp:list -->
<ul class="wp-block-list"><!-- wp:list-item -->
<li>Example text</li>
<!-- /wp:list-item --></ul>
<!-- /wp:list -->"
`;

exports[`Paragraph block transforms to Preformatted block 1`] = `
"<!-- wp:preformatted -->
<pre class="wp-block-preformatted">Example text</pre>
<!-- /wp:preformatted -->"
`;

exports[`Paragraph block transforms to Pullquote block 1`] = `
"<!-- wp:pullquote -->
<figure class="wp-block-pullquote"><blockquote><p>Example text</p></blockquote></figure>
<!-- /wp:pullquote -->"
`;

exports[`Paragraph block transforms to Quote block 1`] = `
"<!-- wp:quote -->
<blockquote class="wp-block-quote"><!-- wp:paragraph -->
<p>Example text</p>
<!-- /wp:paragraph --></blockquote>
<!-- /wp:quote -->"
`;

exports[`Paragraph block transforms to Verse block 1`] = `
"<!-- wp:verse -->
<pre class="wp-block-verse">Example text</pre>
<!-- /wp:verse -->"
`;
